opm/utility/ECLResultData.cpp

```diff
@@ -770,130 +1074,586 @@ namespace Opm {
+    /// \param[in,out] rhs Object from which to constructo new \c Impl
+    ///    instance.  Underlying result-set accessor is null upon return.
+    Impl(Impl&& rhs);
+
+    const ecl_file_type* getRawFilePtr() const;
+
+    /// Query current result-set view for availability of particular named
+    /// result vector in particular enumerated grid.
+    ///
+    /// \param[in] vector Named result vector for which to query data
+    ///    availability.
+    ///
+    /// \param[in] gridID Identity of specific grid for which to query data
+    ///    availability.
+    ///
+    /// \return Whether or not keyword data for the named result vector is
+    ///    available in the specific grid.
+    bool haveKeywordData(const std::string& vector,
+                         const std::string& gridName) const;
+
+    /// Retrieve current result-set view's data values for particular named
+    /// result vector in particular enumerated grid.
+    ///
+    /// Will fail (throw an exception of type std::invalid_argument) unless
+    /// the requested keyword data is available in the specific grid in the
+    /// current active view.
+    ///
+    /// \tparam T Element type of return value.  The underlying keyword data
+    ///    will be converted to this type if needed and possible.  Note that
+    ///    some type combinations do not make sense.  It is, for instance,
+    ///    not possible to retrieve keyword values of an underlying
+    ///    arithmetic type in the form of a \code std::vector<std::string>
+    ///    \endcode.  Similarly, we cannot access underlying character data
+    ///    through elements of an arithmetic type (e.g., \code
+    ///    std::vector<double> \endcode.)
+    ///
+    /// \param[in] vector Named result vector for which to retrieve
+    ///    keyword data.
+    ///
+    /// \param[in] gridID Identity of specific grid for which to
+    ///    retrieve keyword data.
+    ///
+    /// \return Keyword data values.  Empty if type conversion fails.
+    template <typename T>
+    std::vector<T>
+    keywordData(const std::string& vector,
+                const std::string& gridName) const;
+
+private:
+    using SectionID =
+        ECLImpl::InitFileSections::SectionID;
+
+    /// Result of searching for a particular pairing of (vector,gridName) in
+    /// INIT.
+    struct LookupResult
+    {
+        /// In what INIT file section the KW was located (-1 if not found).
+        SectionID sectID;
+
+        /// Local ID, within sectID, of grid section data section that hosts
+        /// the 'vector' (-1 if not found).
+        int gridSectID;
+    };
+
+    /// Pairing of kw vector and grid name.
+    struct KWKey {
+        /// Keyword/result set vector
+        std::string vector;
+
+        /// ID of grid for which to look up 'vector'.
+        std::string gridName;
+    };
+
+    /// Comparator (std::set<> and std::map<>) for KWKeys.
+    struct CompareKWKey {
+        bool operator()(const KWKey& k1, const KWKey& k2)
```

Comment:
Minor bug here: missing const for method. Seems to have been accepted by libstdc++(gcc), but not by libc++(clang).